### PR TITLE
Give nunit.engine and nunit.engine.api assemblies strong names

### PR DESCRIPTION
**Pull request by CharliePoole**
Give these assemblies strong names so they can be referenced by strongly named runners.

### CODE REVIEW

**Comment by rprouse** [comment]
You ended up copying and adding nunit.snk a couple of times. You should add as link so we don't have multiple copies. We already have two and need to clean up.

**Comment by CharliePoole** [comment]
I considered that and checked, finding we already had two. Figured we must have decided to do it that way. :-)

I agree, even though it requires munging the project files.

**Comment by rprouse** [comment]
You don't need to munge the project files, just add existing item, add as
link, then you can select the snk file from the project settings.

**Comment by CharliePoole** [comment]
Cool. That's worth trying just to know about it.

**Comment by CharliePoole** [comment]
In the past, I always edited the csproj file.

**Comment by rprouse** [comment]
I shouldn't have left the extra snk file in the solution. It was one of the
first changes I submitted. I didn't like that there were two, but thought
it must be for a reason so left it.

You could remove the extra with this pr and make the world a better place
;)

**Comment by CharliePoole** [comment]
Can you remind me why we gave the framework tests a strong name?

**Comment by CharliePoole** [comment]
Ah... never mind... because of friend access.

**Comment by CharliePoole** [comment]
I started over on top of the new master and forced the update. This now cleans up the framework key files as well as giving strong names to nunit.engine and nunit.engine.api.

File: src/NUnitFramework/tests/nunit.framework.tests-sl-5.0.csproj [hunk 1]
```diff
@@ -297,7 +297,6 @@
     <Compile Include="Syntax\TestCompiler.cs" />
     <Compile Include="Syntax\ThrowsTests.cs" />
     <Compile Include="Syntax\TypeConstraints.cs" />
-    <Compile Include="TestContextTests.cs" />
```
Comment (rprouse) [inline src/NUnitFramework/tests/nunit.framework.tests-sl-5.0.csproj]: Why was TestContextTests dropped in SL?

Comment (CharliePoole) [inline src/NUnitFramework/tests/nunit.framework.tests-sl-5.0.csproj]: Slip of the mouse. :-) I found it missing in other projects and added the file back, but I didn't notice it was no longer in Silverlight.

Comment (CharliePoole) [inline src/NUnitFramework/tests/nunit.framework.tests-sl-5.0.csproj]: I'll put it back and merge if you think the rest looks OK.

Comment (rprouse) [inline src/NUnitFramework/tests/nunit.framework.tests-sl-5.0.csproj]: Except for that, looks good.

**Comment by CharliePoole** [comment]
Fixed it on GitHub. I'll merge when the CI finishes.

**Pull request merged**